src/framegrab/grabber.py

```diff
@@ -666,9 +669,44 @@ def __init__(self, config: dict):
         thread.daemon = True
         thread.start()
 
+    def _substitute_rtsp_password(self, config: dict) -> dict:
+        """
+        Substitutes the password placeholder in the rtsp_url with the actual password
+        from an environment variable.
+        The URL format should take this format
+            Ex: rtsp://admin:{{RTSP_PASSWORD}}@10.0.0.0/cam/realmonitor?channel=1&subtype=0
+
+        This function looks for a string between {{ and }} to find an environment variable with that name.
+        If the environment variable is found, its value will be substituted in the rtsp_url.
+        NOTE: This can also work for multiple RTSP URLs in the same config file as long
+            as each one has a unique password placeholder.
+        """
+
+        pattern = r"\{\{(.+?)\}\}"  # Regular expression to find {{ENV_VAR}}
```

Comment:
```suggestion
        pattern = r"\{\{([A-Z]+?)\}\}"  # Regular expression to find {{ENV_VAR}}
```

Reply:
I think we're missing an underscore here, so the pattern should be `pattern = r"\{\{([A-Z_]+?)\}\}"`

Reply:
I think this is the right pattern to use here: 

```python
pattern = r"\{\{([A-Z_][A-Z0-9_]*?)\}\}"
```